Errors reported for 3rd-party dependencies

When I run type-checking of my code with `mypy` - everything works correctly.
But, when I try to run type-tests on some module, then it breaks, example:

```
E     ../../../../../../Users/sobolev/Documents/github/returns/.venv/lib/python3.7/site-packages/hypothesis/vendor/pretty:706: error: Call to untyped function "_safe_getattr" in typed context (diff)
```

Same, when I type check my Django project with `mypy` it works fine.

I tried adding this plugin, and running any test fails with:

```
/Users/silviogutierrez/Sites/silviogutierrez/reactivated/master/tests/test_foo.yml:6:
E   pytest_mypy_plugins.utils.TypecheckAssertionError: Invalid output:
E   Expected:
E     main:4: note: Revealed type is 'builtins.str' (diff)
E   Actual:
E     ../../../../../../Users/silviogutierrez/Sites/silviogutierrez/reactivated/master/.venv/lib/python3.8/site-packages/django-stubs/contrib/auth/base_useri:18: error: Need type annotation for 'password' (diff)
E     ../../../../../../Users/silviogutierrez/Sites/silviogutierrez/reactivated/master/.venv/lib/python3.8/site-packages/django-stubs/contrib/auth/base_useri:19: error: Need type annotation for 'last_login' (diff)
E     ../../../../../../Users/silviogutierrez/Sites/silviogutierrez/reactivated/master/.venv/lib/python3.8/site-packages/django-stubs/contrib/auth/modelsi:23: error: Need type annotation for 'name' (diff)
E     ../../../../../../Users/silviogutierrez/Sites/silviogutierrez/reactivated/master/.venv/lib/python3.8/site-packages/django-stubs/contrib/auth/modelsi:24: error: Need type annotation for 'content_type' (diff)
E     ../../../../../../Users/silviogutierrez/Sites/silviogutierrez/reactivated/master/.venv/lib/python3.8/site-packages/django-stubs/contrib/auth/modelsi:25: error: Need type annotation for 'codename' (diff)
E     ../../../../../../Users/silviogutierrez/Sites/silviogutierrez/reactivated/master/.venv/lib/python3.8/site-packages/django-stubs/contrib/auth/modelsi:34: error: Need type annotation for 'name' (diff)
E     ../../../../../../Users/silviogutierrez/Sites/silviogutierrez/reactivated/master/.venv/lib/python3.8/site-packages/django-stubs/contrib/auth/modelsi:35: error: Need type annotation for 'permissions' (diff)
E     ../../../../../../Users/silviogutierrez/Sites/silviogutierrez/reactivated/master/.venv/lib/python3.8/site-packages/django-stubs/contrib/auth/modelsi:57: error: Need type annotation for 'is_superuser' (diff)
E     ../../../../../../Users/silviogutierrez/Sites/silviogutierrez/reactivated/master/.venv/lib/python3.8/site-packages/django-stubs/contrib/auth/modelsi:58: error: Need type annotation for 'groups' (diff)
E     ../../../../../../Users/silviogutierrez/Sites/silviogutierrez/reactivated/master/.venv/lib/python3.8/site-packages/django-stubs/contrib/auth/modelsi:59: error: Need type annotation for 'user_permissions' (diff)
E     ../../../../../../Users/silviogutierrez/Sites/silviogutierrez/reactivated/master/.venv/lib/python3.8/site-packages/django-stubs/contrib/auth/modelsi:70: error: Need type annotation for 'username' (diff)
E     ../../../../../../Users/silviogutierrez/Sites/silviogutierrez/reactivated/master/.venv/lib/python3.8/site-packages/django-stubs/contrib/auth/modelsi:71: error: Need type annotation for 'first_name' (diff)
E     ../../../../../../Users/silviogutierrez/Sites/silviogutierrez/reactivated/master/.venv/lib/python3.8/site-packages/django-stubs/contrib/auth/modelsi:72: error: Need type annotation for 'last_name' (diff)
E     ../../../../../../Users/silviogutierrez/Sites/silviogutierrez/reactivated/master/.venv/lib/python3.8/site-packages/django-stubs/contrib/auth/modelsi:73: error: Need type annotation for 'email' (diff)
E     ../../../../../../Users/silviogutierrez/Sites/silviogutierrez/reactivated/master/.venv/lib/python3.8/site-packages/django-stubs/contrib/auth/modelsi:74: error: Need type annotation for 'is_staff' (diff)
E     ../../../../../../Users/silviogutierrez/Sites/silviogutierrez/reactivated/master/.venv/lib/python3.8/site-packages/django-stubs/contrib/auth/modelsi:75: error: Need type annotation for 'is_active' (diff)
E     ../../../../../../Users/silviogutierrez/Sites/silviogutierrez/reactivated/master/.venv/lib/python3.8/site-packages/django-stubs/contrib/auth/modelsi:76: error: Need type annotation for 'date_joined' (diff)
E     ../../../../../../Users/silviogutierrez/Sites/silviogutierrez/reactivated/master/.venv/lib/python3.8/site-packages/django-stubs/contrib/sites/modelsi:17: error: Need type annotation for 'domain' (diff)
E     ../../../../../../Users/silviogutierrez/Sites/silviogutierrez/reactivated/master/.venv/lib/python3.8/site-packages/django-stubs/contrib/sites/modelsi:18: error: Need type annotation for 'name' (diff)
```

Basically, it seems you can't use this plugin with any 3rd-party *typed* dependency. Be it stubs or just a fully first-party typed library.

--- a/pytest_mypy_plugins/collect.py
+++ b/pytest_mypy_plugins/collect.py
@@ -155,7 +155,7 @@ def pytest_addoption(parser: Parser) -> None:
         "Has to be top-level.",
     )
     group.addoption(
-        "--mypy-only-local-stub", 
-        action="store_true", 
+        "--mypy-only-local-stub",
+        action="store_true",
         help="mypy will ignore errors from site-packages",
     )

--- a/pytest_mypy_plugins/collect.py
+++ b/pytest_mypy_plugins/collect.py
@@ -154,3 +154,8 @@ def pytest_addoption(parser: Parser) -> None:
         help="Fully qualified path to the extension hook function, in case you need custom yaml keys. "
         "Has to be top-level.",
     )
+    group.addoption(
+        "--mypy-only-local-stub", 
+        action="store_true", 
+        help="mypy will ignore errors from site-packages",
+    )

--- a/pytest_mypy_plugins/item.py
+++ b/pytest_mypy_plugins/item.py
@@ -139,6 +139,7 @@ class YamlTestItem(pytest.Item):
         self.additional_mypy_config = mypy_config
         self.parsed_test_data = parsed_test_data
         self.same_process = self.config.option.mypy_same_process
+        self.test_only_local_stub = self.config.option.mypy_only_local_stub
 
         # config parameters
         self.root_directory = self.config.option.mypy_testing_base
@@ -296,11 +297,12 @@ class YamlTestItem(pytest.Item):
     def prepare_mypy_cmd_options(self, execution_path: Path) -> List[str]:
         mypy_cmd_options = [
             "--show-traceback",
-            "--no-silence-site-packages",
             "--no-error-summary",
             "--no-pretty",
             "--hide-error-context",
         ]
+        if not self.test_only_local_stub:
+            mypy_cmd_options.append("--no-silence-site-packages")
         if not self.disable_cache:
             mypy_cmd_options.extend(["--cache-dir", self.incremental_cache_dir])
 

